app/src/test/java/be/xplore/githubmetrics/app/WorkflowRunsIntegrationTest.java

```diff
@@ -0,0 +1,63 @@
+package be.xplore.githubmetrics.app;
+
+import be.xplore.githubmetrics.domain.schedulers.ports.WorkflowRunsUseCase;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.junit5.WireMockTest;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.actuate.observability.AutoConfigureObservability;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.ok;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+
+@ActiveProfiles("test")
+@AutoConfigureMockMvc
+@WireMockTest(httpPort = 8081)
+@SpringBootTest(
+        webEnvironment = RANDOM_PORT
+)
+@AutoConfigureObservability
+class WorkflowRunsIntegrationTest {
+
+    @Autowired
+    private WorkflowRunsUseCase workflowRunsUseCase;
+    @Autowired
+    private MockMvc mockMvc;
+
+    @BeforeEach
+    void setUp() {
+
+        stubFor(WireMock.get("/orgs/github-insights/repos")
+                .willReturn(ok()
+                        .withHeader("Content-Type", "application/json")
+                        .withBodyFile("GithubMetricsRepositoryTestData.json")));
+
+        stubFor(WireMock.get(urlEqualTo("/repos/github-insights/github-metrics/actions/runs?created=%3E%3D2024-04-25"))
```

Comment:
Date should be generated so its always yesterday or whatever. request needs to line up with the actual code

Reply:
done! also relised that the date creation code could be extracted since we are using it in soo many tests. Already created a ticket for it #60